woocommerce/woocommerce
# [ Product Block Editor ] Create Variation options block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Create variation options block and new components.

I copied the components in the attribute-control folder to a new one, called variation-options-control. I am on the fence about doing this right now, since it created a lot of duplicated code. The idea is that attributes and variation options will increase in difference with time, and then this will pay off. As of this time they still look pretty similar. The alternative is to increment the attributes-related components to make them support both scenarios.

I also had to add 'variable' to `supported_post_types` in Init.php, otherwise it redirected to the legacy editor.

Adding, editing and removing variation options also kind of worked out of the box, even though they were not part of the requirements of this issue.

Closes #38882 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Disable the new product editor in WooCommerce > Settings > Advanced > Features.
2. In the legacy product editor, create a variable product, add one local attribute and one global attribute with any terms and with the "Used for variations" checkbox checked. Publish the product.
3. Enable the new product editor in WooCommerce > Settings > Advanced > Features.
4. Using WCA Test Helper, enable the feature "product-variation-management"
5. Open the created product with the new product editor.
6. Go to the Variations tab.
7. Check that the attributes you added in the legacy product editor are there.





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
